spock-core/src/main/java/spock/lang/Snapshot.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package spock.lang;
+
+import org.spockframework.runtime.extension.ExtensionAnnotation;
+import org.spockframework.runtime.extension.builtin.SnapshotExtension;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.nio.charset.Charset;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.FIELD, ElementType.PARAMETER})
+@ExtensionAnnotation(SnapshotExtension.class)
+public @interface Snapshot {
+  /**
+   * The file extension to use for the snapshot files.
+   * Defaults to "txt".
+   */
+  String extension() default "txt";
+
+  /**
+   * The name of the {@link Charset} to use when storing and reading the snapshot.
+   * Defaults to "UTF-8".
+   */
+  String charset() default "UTF-8";
+}
```

Comment:
Could we maybe add an additional property: 

```java 
ELineEndingSanitization sanitizeLineEnding() default ELineEndingSanitization.NL;

enum ELineEndingSanitization {
  NL,
  AS_IS,
  SYSTEM
}
```

Because I see may tests fail for system like this, where windows users write such files but then fail on Linux and vice versa. So I would choose the default to always sanitize it to `NL (\n)`, if nothing was specified.

This would ease the common line ending problem a bit IMHO for people not thinking about it.

Reply:
What is with the `E` prefix, was this meant as analogue to the I-prefix of interfaces?

We already [support line ending normalization](https://github.com/spockframework/spock/pull/1873/files#diff-08417e069ba5fdb0cf22f53da5070bc07221e7a5b5d6ecd88246495d57e23a44R53), which defaults to `\n`. 
I was not keen on adding anything else, but I wanted to leave the door open for users to roll their own if they absolutely want to.

Reply:
Yeah the `E` is a habit in my work projects, and yes it is like the `I` prefix for interfaces. It was just engrained in my brain when typing an enum, that was not meant to "leak" ;).